React SDK local build is not compatible with React 19 types

## Summary

`@webspatial/react-sdk` currently advertises `react >=18.0.0`, but the package's local development/type build path still has React 18-only assumptions. When the SDK is linked into a React 19 app and rebuilt with React 19 type packages, declaration generation fails.

## Reproduction

1. Use a Vite React app with React 19 and `jsxImportSource: "@webspatial/react-sdk"`.
2. Link local `@webspatial/core-sdk` and `@webspatial/react-sdk` packages from this repo into the app.
3. Upgrade `packages/react` dev dependencies to React 19 / `@types/react` 19 and run `pnpm -C packages/react build`.

## Observed

DTS generation fails in `packages/react/src/reality/components/UnlitMaterial.tsx` because React 19 no longer accepts zero-argument `useRef<T>()` in the same way React 18 types did:

```text
src/reality/components/UnlitMaterial.tsx(17,23): error TS2554: Expected 1 arguments, but got 0.
src/reality/components/UnlitMaterial.tsx(39,7): error TS2322: Type 'undefined' is not assignable to type 'SpatialUnlitMaterial'.
```

`ModelAsset.tsx` has the same zero-argument `useRef<T>()` pattern.

## Expected

The React SDK should continue supporting React 18 while also building cleanly against React 19 types, matching the package peer dependency range.

--- a/.changeset/calm-react-compat.md
+++ b/.changeset/calm-react-compat.md
@@ -1,0 +1,5 @@
+---
+'@webspatial/react-sdk': patch
+---
+
+Fix React 19 type compatibility for local builds by initializing internal `useRef` values explicitly while preserving React 18 support.

--- a/packages/react/src/reality/components/ModelAsset.tsx
+++ b/packages/react/src/reality/components/ModelAsset.tsx
@@ -19,7 +19,7 @@ const resolveAssetUrl = (url: string): string => {
 
 export const ModelAsset: React.FC<Props> = ({ children, ...options }) => {
   const ctx = useRealityContext()
-  const materialRef = useRef<SpatialModelAsset>()
+  const materialRef = useRef<SpatialModelAsset | undefined>(undefined)
   useEffect(() => {
     const controller = new AbortController()
     if (!ctx) return

--- a/packages/react/src/reality/components/UnlitMaterial.tsx
+++ b/packages/react/src/reality/components/UnlitMaterial.tsx
@@ -14,7 +14,7 @@ export const UnlitMaterial: React.FC<UnlitMaterialProps> = ({
   ...options
 }) => {
   const ctx = useRealityContext()
-  const materialRef = useRef<SpatialUnlitMaterial>()
+  const materialRef = useRef<SpatialUnlitMaterial | undefined>(undefined)
   const isInitializedRef = useRef(false)
 
   useEffect(() => {

